### PR TITLE
Fix linter rule docs using `:` instead of `/` as separator

### LINKED ISSUE #2520: Docs on creating a linter use ":" delimiter when it should be "/".
In the docs for linters

https://microsoft.github.io/typespec/extending-typespec/linters

there are a few places that use a ":" delimiter that should be a "/". In particular
```
        enable: { [`@typespec/my-linter:${requiredDocRule.name}`]: true },
```
should be
```
        enable: { [`@typespec/my-linter/${requiredDocRule.name}`]: true },
```

And the same mistake appears in the linters section of the Configuration docs:

https://microsoft.github.io/typespec/introduction/configuration#linter---configuring-linters

```
linter:
  extends: # Extend `recommended` ruleset from @typespec/best-practices library
    - "@typespec/best-practices:recommended"
```
should be
```
linter:
  extends: # Extend `recommended` ruleset from @typespec/best-practices library
    - "@typespec/best-practices/recommended"
```

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #2520

### CODE REVIEW

File: docs/extending-typespec/linters.md [hunk 1]
```diff
@@ -109,10 +109,10 @@ export const $lib = createTypeSpecLibrary({
...
         // (optional) A ruleset takes a map of rules to explicitly enable
-        enable: { [`@typespec/my-linter:${requiredDocRule.name}`]: true },
+        enable: { [`@typespec/my-linter/${requiredDocRule.name}`]: true },
 
         // (optional) A rule set can extend another rule set
-        extends: ["@typespec/best-practices:recommended"],
+        extends: ["@typespec/best-practices/recommended"],
 
         // (optional) A rule set can disable a rule enabled in a ruleset it extended.
         disable: {
```
Comment (mikekistler) [inline docs/extending-typespec/linters.md:118]: I think the following lines also need changing: 119, 127, 150

Comment (timotheeguerin) [reply docs/extending-typespec/linters.md:118]: yep, missed that...

Comment (mikekistler) [reply docs/extending-typespec/linters.md:118]: Also need to fix linter config (~line 315) in `docs/introduction/configuration/configuration.md`

**Review by mikekistler — APPROVED** [review]
Looks good! 👍

**Pull request merged**